tests/integration/conftest.py

```diff
@@ -58,3 +72,34 @@ def bans_url_1(configured_httpserver):
 @pytest.fixture
 def bans_url_2(configured_httpserver):
     return configured_httpserver.url_for("/bansummary-changed.text")
+
+
+@pytest.fixture
+def sequential_handler(request):
+    if hasattr(request, "param"):
+        responses = request.param
+    else:
+        responses = []
+        # Any test that requests this fixture should perform indirect parameterization to set the sequence of expected responses.
+
+    # Create a generator out of sequence of respnoses
+    resp_generator = (v for v in responses)
+
+    # Handler returns next item in generator, and state is maintained until fixture is destroyed
+    def my_handler(req):
+        return next(resp_generator)
+
+    return my_handler
```

Comment:
This guy uses the behaviour of python generators to provide a handler function which returns the next item in a given sequence whenever invoked.

Reply:
Because this handler is provided as a fixture, it gets destroyed whenever the fixture goes out of scope. This allows for the sequence to be "reset" between tests (this having the default fixture scope of "function")

Reply:
What would `request.param` typically be?